src/cloudant/_common_util.py

```diff
@@ -263,6 +263,20 @@ def get_docs(r_session, url, encoder=None, headers=None, **params):
     resp.raise_for_status()
     return resp
 
+#pylint: disable=unused-argument
+def append_response_error_content(response, **kwargs):
+    """
+    Provides a helper to add HTTP response error and reason messages.
+    """
+    try:
+        if (response.status_code >= 400 and response.json() and
+                ('error' in response.json()) and ('reason' in response.json())):
+            resp_content = response.json()
+            response.url += ' %s %s' % (resp_content['error'], resp_content['reason'])
```

Comment:
I understand why this is being done via the url property at the moment, but I think it could be risky. I'm concerned about modifying it in case anything else tries to use the `url` (e.g. to form a follow on request to the same URL). I don't know exactly how python-requests handles it's retries and I expect the retry logic gets called before our hook anyway, but if anything calls `response.url` expecting something valid then this type of modification could cause problems.

I think we might be better modifying the `response.reason` instead of the URL.